File: core/src/com/inkostilation/pong/commands/response/ResponseFieldCommand.java
package com.inkostilation.pong.commands.response;


import com.inkostilation.pong.commands.AbstractResponseCommand;
import com.inkostilation.pong.engine.Field;

public class ResponseFieldCommand extends AbstractResponseCommand {

    private Field field;

    public ResponseFieldCommand(Field field) {
        this.field = field;
    }

    @Override
    public void execute()
    {
        getNotifier().notifyObservers(field);
        getNotifier().notifyObservers(field.getPaddle1());
        getNotifier().notifyObservers(field.getPaddle2());
        getNotifier().notifyObservers(field.getBall());
    }
}

File: core/src/com/inkostilation/pong/commands/response/ResponseFieldCommand.kt
package com.inkostilation.pong.commands.response

import com.inkostilation.pong.commands.AbstractResponseCommand
import com.inkostilation.pong.engine.Field

class ResponseFieldCommand(private val field: Field) : AbstractResponseCommand() {
    override fun execute() {
        notifier.notifyObservers(field)
        notifier.notifyObservers(field.paddle1)
        notifier.notifyObservers(field.paddle2)
        notifier.notifyObservers(field.ball)
    }

}